fhir-server/src/main/java/org/linuxforhealth/fhir/server/util/RestAuditLogger.java

```diff
@@ -746,6 +774,142 @@ protected static AuditLogEntry populateAuditLogEntry(AuditLogEntry entry, HttpSe
+            Date startTime, Date endTime, Response.Status responseStatus) { 
+        final String METHODNAME = "populateBundleAuditLogEntry";
+        log.entering(CLASSNAME, METHODNAME);
+
+        // call populateAuditLogEntry to populate common attributes to all rest
+        populateAuditLogEntry(entry, request, resource, startTime, endTime, responseStatus);
+        if (HTTPReturnPreference.REPRESENTATION.equals(FHIRRequestContext.get().getReturnPreference())) {
+            return entry;
+        }
+        if (responseEntry.getResponse() == null || responseEntry.getResponse().getLocation() == null) {
+            return entry;
+        }
+        // Populate the resource id, resource type and version id from the
+        // Bundle response entry location when the return preference is "OperationOutcome".
+        String location = responseEntry.getResponse().getLocation().getValue();
+        String[] parts = location.split("/");
+        if (parts.length > 3) {
+            Collections.reverse(Arrays.asList(parts));
+            entry.getContext().setData(Data.builder().build());
+            entry.getContext().getData().setResourceType(parts[3]);
+            entry.getContext().getData().setVersionId(parts[0]);
+            entry.getContext().getData().setId(parts[2]);
+        }
+        log.exiting(CLASSNAME, METHODNAME);
+        return entry;
+    }
+    
+    /**
+     * Log an audit entry for each deleted resource in a Bundle.Entry
+     * @param auditLogSvc
+     *  The internal FHIR Server API for audit logging.
+     * @param responseEntry
+     *  The Bundle response entry.
+     * @param request
+     *  The HttpServletRequest representation of the REST request.
+     * @param resource
+     *  The Resource object.
+     * @param startTime
+     *  The start time of the request execution.
+     * @param endTime
+     *  The end time of the request execution.
+     * @param responseStatus
+     *  The response status.
+     * @param location
+     * @throws Exception
+     */
+    protected static void logBundleDelete(AuditLogService auditLogSvc, Bundle.Entry responseEntry, HttpServletRequest request, Resource resource,
+        Date startTime, Date endTime, Response.Status responseStatus, String location, String description) throws Exception {
+        final String METHODNAME = "logBundleDelete";
+        log.entering(CLASSNAME, METHODNAME);
+        
+        // Populate the resource id, resource type and version id from the
+        // Bundle response entry extensions
+        for (Extension extension : responseEntry.getExtension()) {
+             AuditLogEntry entry = initLogEntry(AuditLogEventType.FHIR_BUNDLE);
+             populateAuditLogEntry(entry, request, resource, startTime, endTime, responseStatus);
+             String resourceInfo = extension.getValue().as(FHIR_STRING).getValue();
```

Comment:
I think we should double-check the URL of the extension instead of just assuming all extensions will be the ones we're expecting